src/backend/oracle-data-api/src/test/java/ca/bc/gov/open/jag/tco/oracledataapi/util/RandomUtil.java

```diff
@@ -200,53 +212,242 @@ public static EmailHistory createEmailHistory() {
+		return RandomStringUtils.randomAlphabetic(count);
+	}
+
+	/**
+	 * Creates a random string whose length is the number of characters specified. Characters will be chosen from the set of numeric characters.
+	 */
+	public static String randomNumeric(int count) {
+		return RandomStringUtils.randomNumeric(count);
+	}
+
+	/**
+	 * Creates a random string whose length is the number of characters specified.
+	 * Characters will be chosen from the set of Latin alphabetic characters (a-z, A-Z) and the digits 0-9.
+	 */
+	public static String randomAlphanumeric(int count) {
+		return RandomStringUtils.randomAlphanumeric(count);
+	}
+
+	/**
+	 * Returns a random date within the last year.
+	 */
+	public static Date randomDate() {
+		return DateUtils.ceiling(randomDateTime(), Calendar.DAY_OF_MONTH);
+	}
+
+	/**
+	 * Returns a random date within the last year.
+	 */
+	public static Date randomDateTime() {
+		return randomDate(DateUtils.addYears(new Date(), -1), new Date());
+	}
+
+	public static YesNo randomYN() {
+		return (new Random().nextBoolean()) ? YesNo.Y : YesNo.N;
+	}
+
+
+	/**
+	 * Creates a Dispute where every field is populated with random data to the maximum character limit appropriate for each field.
+	 */
+	public static Dispute createFullyPopulatedDispute() {
+		Dispute dispute = new Dispute();
+
+		dispute.setAddressCity(randomCity()); // TODO: what is the max length of this field?
+		dispute.setAddressLine1(randomAlphanumeric(100)); // FIXME: Column length disparity - ORDs is 100, H2 is 500
+		dispute.setAddressLine2(randomAlphanumeric(100)); // FIXME: Column length disparity - ORDs is 100, H2 is 500
+		dispute.setAddressLine3(randomAlphanumeric(100)); // FIXME: Column length disparity - ORDs is 100, H2 is 500
+		dispute.setAddressProvince(randomAlphabetic(30));
+		dispute.setCourtLocation(randomAlphabetic(150));
+		dispute.setDetachmentLocation(randomAlphabetic(150));
+		dispute.setDisputantBirthdate(randomDate());
+		dispute.setDisputantClientId(null);
+		dispute.setDisputantComment(randomAlphabetic(4000));
+		dispute.setDisputantDetectedOcrIssues(randomYN());
+		dispute.setDisputantGivenName1(randomAlphabetic(30));
+		dispute.setDisputantGivenName2(randomAlphabetic(30));
+		dispute.setDisputantGivenName3(randomAlphabetic(30));
+		dispute.setDisputantOcrIssues(randomAlphabetic(500));
+		dispute.setDisputantOrganization(randomAlphabetic(150));
+		dispute.setDisputantSurname(randomAlphabetic(30));
+		dispute.setDisputeCounts(createDisputeCounts(dispute));
+		dispute.setDisputeStatusType(createDisputeStatusType());
+		dispute.setDriversLicenceNumber(randomAlphanumeric(30));
+		dispute.setDriversLicenceProvince(randomAlphabetic(30));
+		dispute.setEmailAddress(randomAlphanumeric(100));
+		dispute.setFilingDate(randomDate());
+		dispute.setFineReductionReason(randomAlphabetic(500));
+		dispute.setHomePhoneNumber(randomNumeric(20));
+		dispute.setInterpreterLanguage(randomAlphabetic(10)); // TODO: what is the max length of this field?

```

Comment:
This field still needs to be created in OCCAM table.

Reply:
Actually, Elaine found out that this is already in OCCAM, but under the field called "languageCd" with a field length of 3.  This change will be a different PR for TCVP-1921